src/main/java/com/sugarcrm/candybean/candybeanRunner/CandybeanRunner.java

```diff
@@ -0,0 +1,332 @@
+	protected final int timeout = Integer.parseInt(candybean.config.getValue("runner.timeout", "0"));
+	protected final int cleanupTimeout = Integer.parseInt(
+			candybean.config.getValue("runner.cleanupTimeout", String.valueOf(timeout)));
+	protected final boolean rerunIfTimedOut = Boolean.parseBoolean(candybean.config.getValue("runner.rerunIfTimedOut", "false"));
+	protected final boolean runBeforeIfTimedOut = Boolean.parseBoolean(candybean.config.getValue("runner.runBeforeIfTimedOut", "true"));
+	protected final boolean runAfterIfTimedOut = Boolean.parseBoolean(candybean.config.getValue("runner.runAfterIfTimedOut", "true"));
+	final boolean INTERRUPT = true;
+
+	/**
+	 * Creates a BlockJUnit4ClassRunner to run klass
+	 *
+	 * @param klass The class to initialize
+	 * @throws InitializationError
+	 */
+	public CandybeanRunner(Class<?> klass) throws InitializationError, CandybeanException{
+		super(klass);
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void run(final RunNotifier notifier) {
+		/* Construct statement that when evaluated:
+		 *  Runs @BeforeClass
+		 *  Runs All children tests with runChild
+		 *  Runs @AfterClass
+		 *
+		 * If the test fails, it notifies the test runner
+		 *
+		 * This method is called on each test class. Evaluating
+		 * the statement then runs runChild on each test within
+		 * the class.
+		 */
+		final Statement statement = classBlock(notifier);
+		try {
+			statement.evaluate();
+		} catch (StoppedByUserException e) {
+			throw e;
+		} catch (Throwable e) {
+			notifier.fireTestFailure(new Failure(getDescription(), e));
+		}
+	}
+
+	/**
+	 * Runs each test using retries and timeouts. The general idea of the this
+	 * method is that it constructs a Callable that when run, runs the test. It
+	 * attempts to retrieve the results of the callable after a certain timeout.
+	 * If the test wasn't ready, or the test failed, it reruns the callable if
+	 * applicable up to some specified number of times.
+	 *
+	 * @param method The test method to run
+	 * @param notifier The test notifier to update
+	 */
+	@Override
+	protected void runChild(final FrameworkMethod method, RunNotifier notifier) {
+		final Description description = describeChild(method);
+
+		// Check the annotations, it the test is marked ignore, ignore it.
+		// If you wanted add a check for custom annotations, perhaps you
+		// wanted an @DoNotReRun annotation, you would check for it here
+		if (method.getAnnotation(Ignore.class) != null) {
+			notifier.fireTestIgnored(description);
+			return;
+		}
+
+		// Get a statement representing a single test
+		final Statement statement = methodBlock(method);
+
+		// Construct a callable to run a single test
+		final Callable<Void> runTest = new Callable<Void>() {
+			public Void call() throws Exception {
+				try {
+					statement.evaluate();
+				} catch (Error | Exception e) {
+					throw e;
+				} catch (Throwable e) {
+					throw new Exception(e);
+				}
```

Comment:
statement.evaluate throws a Throwable. However, Throwable, is subclassed by Error and Exception, and if the Throwable is an instance of one of those, it's preferable to throw that. If it is not, we wrap it in a new Exception and throw that. 

I do agree that it could use a clarifying comment.